scripts/ci/lib.sh

```diff
@@ -819,6 +819,74 @@ send_slack_notice_for_vuln_check_failure() {
     curl -XPOST -d @- -H 'Content-Type: application/json' "$webhook_url"
 }
 
+generate_genesis_dump() {
```

Comment:
it seems like we do this unconditionally. This should use the same conditionals as the others below. No need to run this in PRs which don't ask for it

Reply:
The rest of the workflow depends on the artifact this function produces. If we make this conditional, I think we'd need to supply some sort of `genesis-dump.zip` and `vuln-dump`. Would you want to grab those from a previous run or from the latest `master` run?

fwiw, it looks like we used to do this unconditionally in OSCI

Reply:
In OSCI we halt certain jobs based on this gating JSON: https://github.com/stackrox/scanner/blob/master/scripts/ci/gate-jobs-config.json

Reply:
Oh, I see. Would this mean that images aren't generated/pushed unless they have the `generate-dumps-on-pr` label? Is that ok?

Reply:
this step can take a while, so in PRs we don't generate the genesis dump (unless a specific label is specified). instead, we pull the dumps from GCP. I think this can be seen in get_genesis_dump()